akka-http-core/src/main/scala/akka/http/engine/client/HostConnectionPoolSlot.scala

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.client
+
+import language.existentials
+import java.net.InetSocketAddress
+import scala.util.{ Failure, Success }
+import scala.collection.immutable
+import akka.actor._
+import akka.http.model.{ HttpResponse, HttpRequest }
+import akka.http.util._
+import akka.stream.impl.{ SubscribePending, ExposedPublisher, ActorProcessor }
+import akka.stream.actor._
+import akka.stream.scaladsl._
+import akka.stream._
+
+private object HostConnectionPoolSlot {
+  import HostConnectionPoolGateway.{ RequestContext, ResponseContext }
+
+  sealed trait ProcessorOut
+  final case class ResponseDelivery(response: ResponseContext) extends ProcessorOut
+  sealed trait SlotEvent extends ProcessorOut
+  sealed trait SimpleSlotEvent extends SlotEvent
+  object SlotEvent {
+    final case class RequestCompleted(slotIx: Int) extends SimpleSlotEvent
+    final case class Disconnected(slotIx: Int, failedRequests: Int) extends SimpleSlotEvent
+    final case class RetryRequest(rc: RequestContext) extends SlotEvent
+  }
+
+  case class Ports(
+    requestContextIn: Inlet[RequestContext],
+    responseOut: Outlet[ResponseContext],
+    slotEventOut: Outlet[SlotEvent]) extends Shape {
+
+    override val inlets = requestContextIn :: Nil
+    override def outlets = responseOut :: slotEventOut :: Nil
+
+    override def deepCopy(): Shape = Ports(
+      new Inlet(requestContextIn.toString),
+      new Outlet(responseOut.toString),
+      new Outlet(slotEventOut.toString))
+
+    override def copyFromPorts(inlets: immutable.Seq[Inlet[_]], outlets: immutable.Seq[Outlet[_]]): Shape =
+      Ports(
+        inlets.head.asInstanceOf[Inlet[RequestContext]],
+        outlets.head.asInstanceOf[Outlet[ResponseContext]],
+        outlets.last.asInstanceOf[Outlet[SlotEvent]])
+  }
```

Comment:
Yes, good point. I'll remove the `Ports` here altogether and switch to `FanOutShape2`.